src/coreComponents/mesh/generators/CellBlockManagerABC.hpp

```diff
@@ -191,6 +191,22 @@ class CellBlockManagerABC : public dataRepository::Group
    * @return A reference to constant map.
    */
   virtual std::map< string, SortedArray< localIndex > > const & getNodeSets() const = 0;
+
+  /**
+   * @brief Generates in place the high-order maps for this cell block manager.
+   * @param[in] order The order of the discretization.
+   * @param[in] maxVertexGlobalID: maximum globalID for nodes
+   * @param[in] maxEdgeGlobalID: maximum globalID for edges
+   * @param[in] maxFaceGlobalID: maximum globalID for faces
+   * @param[in] edgeLocalToGlobal The local to global map for edges.
+   * @param[in] faceLocalToGlobal The local to global map for faces.
+   */
```

Comment:
yes, thanks Thomas for the question. The *LocalToGlobal maps (`edgeLocalToGlobal, faceLocalToGlobal, and elementLocalToGlobal`) at the `CellBlockManager::generateHighOrderMaps` function are used to build the `nodeLocalToGlobalNew` map.

The edgeLocalToGlobal, faceLocalToGlobal, and elementLocalToGlobal maps are the same for base mesh-level and high-order mesh-levels, however the nodes local-to-global map will change because we are adding new nodes. The new nodes include: the new nodes on each of the faces + the new nodes on each of the edges + the new nodes on the internal of each element.  

So, the general idea in this function for building nodes local-to-global map for the high-order meshes are: 
1. we keep the globalIDs for all the nodes that in the base mesh-level; 
2. assign the global ID for the new nodes on the edges (according to the global information for the edges from edgeLocalToGlobal map); 
3. assign the global ID for the new nodes on the faces (according to the global information for the faces from faceLocalToGlobal map); 
4. assign the global ID for the new nodes on the internal of each elements (according to the global information for the elements from elementLocalToGlobal map). 

And that's way we need the local-to-global maps information. I hope the explanation is clear enough, please free feel to let me know if there is any question.

Reply:
Hello, thanks for the informations.
Can you copy you explanation into the code (were you think it's relevant) so it does not get lost?